pkg/volume/csi/csi_mounter.go

```diff
@@ -322,11 +312,21 @@ func (c *csiMountMgr) SetUpAt(dir string, mounterArgs volume.MounterArgs) error
 		return err
 	}
 
+	defer func() {
```

Comment:
Add a defer func to make sure that the error is wrapped by an uncertain error.

Reply:
IMO this code is bit tad too clever for. I am not even sure, why we are doing this. 

The main idea is - volume file (and directory possibly) should be removed only if:

- CSI RPC call failed with a final error.
- If saving of file itself fails (because in that case, nothing will get written anyways).

Shouldn't something simple like - https://github.com/gnufied/kubernetes/commit/3bfd3a303f2e4720ad58b7e931f1184a2b9a9407

Reply:
> I am not even sure, why we are doing this.

We know that the volume has been mounted, once the following code fails, it must be return an uncertain error in order to make sure that the kubelet can unmount the volume. so the developers and reviewers must be careful to add/review new codes. A mistaken may lead to a volume in the mounted state forever if it return an unwrapped error and the pod is deleted at the same time.

Reply:
> Shouldn't something simple like - https://github.com/gnufied/kubernetes/commit/3bfd3a303f2e4720ad58b7e931f1184a2b9a9407

reverted my changes and then cherry-pick from this commit.